components/admin_console/data_retention_settings/data_retention_settings.tsx

```diff
@@ -0,0 +1,587 @@
+                                />
+                            </Menu>
+                        </MenuWrapper>
+                    ),
+                },
+                onClick: () => {
+                    browserHistory.push(`/admin_console/compliance/data_retention_settings/custom_policy/${policy.id}`);
+                },
+            };
+        });
+    };
+    private loadPage = async (page: number) => {
+        this.setState({customPoliciesLoading: true});
+        await this.props.actions.getDataRetentionCustomPolicies(page);
+        this.setState({page, customPoliciesLoading: false});
+    }
+    componentDidMount = async () => {
+        await this.loadPage(this.state.page);
+    }
+
+    private nextPage = () => {
+        this.loadPage(this.state.page + 1);
+    }
+
+    private previousPage = () => {
+        this.loadPage(this.state.page - 1);
+    }
+
+    public getPaginationProps = (): {startCount: number; endCount: number; total: number} => {
+        const {page} = this.state;
+        const startCount = (page * PAGE_SIZE) + 1;
+        const total = this.props.customPoliciesCount;
+        let endCount = (page + 1) * PAGE_SIZE;
+        endCount = endCount > total ? total : endCount;
+
+        return {startCount, endCount, total};
+    }
+
+    showEditJobTime = (value: boolean) => {
+        this.setState({showEditJobTime: value});
+    }
+
+    componentDidUpdate = (prevProps: Props, prevState: State) => {
+        if (prevState.showEditJobTime !== this.state.showEditJobTime && this.state.showEditJobTime) {
+            this.inputRef.current?.focus();
+        }
+    }
+
+    handleCreateJob = async (e?: React.SyntheticEvent) => {
+        e?.preventDefault();
+        const job = {
+            type: JobTypes.DATA_RETENTION as JobType,
+        };
+
+        await this.props.actions.createJob(job);
+        await this.props.actions.getJobsByType(JobTypes.DATA_RETENTION as JobType);
+    };
+
+    getJobTimeOptions = (): OptionType[] => {
+        const minuteIntervals = ['00', '15', '30', '45'];
+        const options = [];
+        for (let i = 0; i < minuteIntervals.length; i++) {
+            options.push({label: `12:${minuteIntervals[i]}am`, value: `00:${minuteIntervals[i]}`});
+        }
+        for (let h = 1; h < 24; h++) {
+            let hourLabel = h;
+            let hourValue = `${h}`;
+            const timeOfDay = h >= 12 ? 'pm' : 'am';
+            if (hourLabel < 10) {
+                hourValue = `0${hourValue}`;
+            }
+            if (hourLabel > 12) {
+                hourLabel -= 12;
+            }
+            for (let i = 0; i < minuteIntervals.length; i++) {
+                options.push({label: `${hourLabel}:${minuteIntervals[i]}${timeOfDay}`, value: `${hourValue}:${minuteIntervals[i]}`});
+            }
+        }
+
+        return options;
+    }
+
+    changeJobTimeConfig = async (value: string) => {
+        const newConfig = JSON.parse(JSON.stringify(this.props.config));
```

Comment:
Non blocking question. Can we have a util method for deep extend? Example Deep Extend section in http://youmightnotneedjquery.com.

Reply:
Yeah I could add something like that in the future but this is ok for now I think